Source/VMware.vSphereDSC/Tests/Integration/VMHostSyslog.Integration.Tests.ps1

```diff
@@ -0,0 +1,183 @@
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Name,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Server,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $User,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Password
+)
+
+$script:dscResourceName = 'VMHostSyslog'
+$script:dscConfig = $null
+$script:moduleFolderPath = (Get-Module VMware.vSphereDSC -ListAvailable).ModuleBase
+$script:integrationTestsFolderPath = Join-Path (Join-Path $moduleFolderPath 'Tests') 'Integration'
+$script:configurationFile = "$script:integrationTestsFolderPath\Configurations\$($script:dscResourceName)\$($script:dscResourceName)_Config.ps1"
+
+$script:configWithDefaultSettings = "$($script:dscResourceName)_WithDefaultSettings_Config"
+$script:configWithNotDefaultSettings = "$($script:dscResourceName)_WithNotDefaultSettings_Config"
+
+$script:connection = Connect-VIServer -Server $Server -User $User -Password $Password
+$script:vmHost = $null
+
+$script:LogHost = 'udp://vli.eng.vmware.com:516'
+$script:LogHost2 = 'udp://vli2.eng.vmware.com:516'
+$script:CheckSslCerts = $true
+$script:DefaultRotate = 10
+$script:DefaultSize = 100
+$script:DefaultTimeout = 180
+$script:DropLogRotate = 10
+$script:DropLogSize = 100
+$script:Logdir = '/scratch/log'
+$script:Logdir2 = '/scratch/log2'
+$script:LogdirUnique = $false
+$script:QueueDropMark = 90
+
+$script:resource_WithDefaultSettings = @{
+    Name = $Name
+    Server = $Server
+    LogHost = $script:LogHost
+    CheckSslCerts = $script:CheckSslCerts
+    DefaultRotate = $script:DefaultRotate
+    DefaultSize = $script:DefaultSize
+    DefaultTimeout = $script:DefaultTimeout
+    DropLogRotate = $script:DropLogRotate
+    DropLogSize = $script:DropLogSize
+    Logdir = $script:Logdir
+    LogdirUnique = $script:LogdirUnique
+    QueueDropMark = $script:QueueDropMark
+}
+
+$script:resource_WithNotDefaultSettings = @{
+    Name = $Name
+    Server = $Server
+    LogHost = $script:LogHost1
+    CheckSslCerts = -not $script:CheckSslCerts
+    DefaultRotate = $script:DefaultRotate + 1
+    DefaultSize = $script:DefaultSize + 1
+    DefaultTimeout = $script:DefaultTimeout + 1
+    DropLogRotate = $script:DropLogRotate + 1
+    DropLogSize = $script:DropLogSize + 1
+    Logdir = $script:Logdir2
+    LogdirUnique = -not $script:LogdirUnique
+    QueueDropMark = $script:QueueDropMark + 1
+}
+
+. $script:configurationFile -Name $Name -Server $Server -User $User -Password $Password
+
+$script:mofFileWithDefaultSettings = "$script:integrationTestsFolderPath\$($script:configWithDefaultSettings)\"
+$script:mofFileWithNotDefaultSettings = "$script:integrationTestsFolderPath\$($script:configWithNotDefaultSettings)\"
+
+Describe "$($script:dscResourceName)_Integration" {
+    Context "When using configuration $($script:configWithDefaultSettings)" {
+        It 'Should compile and apply the MOF without throwing' {
+            # Assert
+            { $startDscConfigurationParameters = @{
```

Comment:
Here we should have the following: Start-DscConfiguration @startDscConfigurationParameters | Should -Not -Throw ( And for all other Start-DscConfiguration Tests the same applies).